nixos/modules/services/misc/freeswitch.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ...}:
+with lib;
+let
+  cfg = config.services.freeswitch;
+  pkg = cfg.package;
+  configDirectory = pkgs.runCommand "freeswitch-config-d" { } ''
+    mkdir -p $out
+    cp -r ${pkg}/etc/freeswitch/* $out
+    chmod -R +w $out
+    ${concatStringsSep "\n" (mapAttrsToList (file: text: ''
+      mkdir -p $out/$(dirname ${file})
+      cat >$out/${file} <<'EOF'
+      ${text}
+      EOF
+    '') cfg.configDir)}
+  '';
+  configPath = if cfg.enableReload
+    then "/etc/freeswitch"
+    else configDirectory;
+in {
+  options = {
+    services.freeswitch = {
+      enable = mkEnableOption "FreeSWITCH";
+      enableReload = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Reload FreeSWITCH when configuration directory changes (instead of restart).
+          The configuration directory is exposed at <filename>/etc/freeswitch</filename>.
+          See also <literal>systemd.services.*.restartIfChanged</literal>.
+        '';
+      };
+      configDir = mkOption {
+        type = types.attrsOf types.lines;
+        default = { };
+        description = ''
+          Override file in FreeSWITCH config directory.
+          Each top-level attribute denotes a file path in the configuration directory, its value is the file content.
```

Comment:
I suggest to change this option to `attrsOf path`. Then you can still set raw strings with `"freeswitch.xml" = pkgs.writeText "foo" "..."`, but it also allows you to do produce the string at build-time, which you can use to do the substitution in the test.

Reply:
Also: A link to the manual where these files are described would be helpful

Reply:
Done. This way it's better i think, thanks!